### PR TITLE
Fix stitched ciphersuites in TLS1.0

### PR DESCRIPTION
**Pull request by mattcaswell**
TLS1.0 does not have an explicit IV in the record, and therefore we should
not attempt to remove it.

### CODE REVIEW

File: providers/implementations/ciphers/cipher_aes_cbc_hmac_sha.c [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 /* Dispatch functions for AES_CBC_HMAC_SHA ciphers */
 
+/* Only for SSL3_VERSION and TLS1_VERSION */
```
Comment (kaduk) [inline providers/implementations/ciphers/cipher_aes_cbc_hmac_sha.c:19]: I'd almost rather hardcode the literal constants for the versions (with a comment) than pull in `ssl.h` here...

Comment (mattcaswell) [reply providers/implementations/ciphers/cipher_aes_cbc_hmac_sha.c:19]: Hmmm. Really? I'm not sure that is better?

**Comment by mattcaswell** [comment]
> Are the "_hw" versions of these ciphers also impacted?

The "_hw" versions aren't really different things.  There just different source files that go together with this one to form the complete cipher implementation. So the change here should be sufficient.

**Comment by kaduk** [comment]
Thanks for the responses, Matt -- please go ahead and merge as-is; my comments have been addressed.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**